src/components/RightSidebar.tsx

```diff
@@ -0,0 +1,58 @@
+import { ReactNode, useState } from "react";
+import { Css } from "src/Css";
+import { useTestIds } from "src/utils";
+import { IconKey } from "./Icon";
+import { IconButton } from "./IconButton";
+
+export type SidebarContentProps = {
+  icon: IconKey;
+  render: () => ReactNode;
+};
+
+export type RightSidebarProps = {
+  content: SidebarContentProps[];
+};
+
+export function RightSidebar({ content }: RightSidebarProps) {
+  const [selectedIcon, setSelectedIcon] = useState<IconKey | undefined>(undefined);
+  const tid = useTestIds({}, "rightSidebar");
+
+  return (
+    <>
+      <div css={Css.df.jcfe.relative.pr3.$}>
+        <div css={Css.df.gap2.z1.$}>
+          {content.map(({ icon }) => (
+            <IconButton
+              key={`${icon}-${selectedIcon}`}
+              circle
+              active={icon === selectedIcon}
+              onClick={() => setSelectedIcon(icon)}
+              icon={icon}
+              inc={3.5}
+            />
+          ))}
+        </div>
+      </div>
+      <div
+        css={{
+          ...Css.wPx(selectedIcon ? 380 : 0).relative.topPx(-48).z0.$,
+          transition: selectedIcon ? "transform 300ms ease-in" : "transform 300ms ease-out",
+          transform: selectedIcon ? "translateX(0)" : "translateX(100%)",
```

Comment:
I attempted adding a sidebar transition. It only works when the sidebar is displayed but not when it's hidden because the sidebar content is hidden immediately when the x icon is clicked. I didn't quite figure this out within the timebox I set for myself, but please give input if you have a better way to set up this transition.

Reply:
update this to use framer motion and it's working better now